lib/jbuild

```diff
@@ -0,0 +1,51 @@
+(* -*- tuareg -*- *)
+#require "unix"
+
+let configuration =
+  let ic = open_in "../config.mk" in
+  let read_lines io =
+    let rec loop acc =
+      match input_line io with
+      | l -> loop (l :: acc)
+      | exception End_of_file -> List.rev acc
+    in
+    loop []
+  in
+  let lines = read_lines ic in
+  close_in ic;
+
+  let convert_config_line x =
+    match String.split_on_char '=' x with
+    (* note we need to include escape backslashes in the string to be
+     passed in the ksprintf function below so it can be echoed correctly *)
+    | [a;b] -> Printf.sprintf "let %s=\\\"%s\\\"" a b
+    | _     -> ""
+  in
+  lines
+  |> List.map convert_config_line
+  |> List.filter (fun x -> x <> "")
+  |> String.concat "\n"
+
+
+let () = Printf.ksprintf Jbuild_plugin.V1.send {|
+(jbuild_version 1)
+
+(rule (
+  (targets (xcp_inventory_config.ml))
+  (deps ())
+  (action (
+    with-stdout-to ${@} (echo "%s")
```

Comment:
Can't we copy `config.mk` to `xcp_inventory_config.ml` instead of doing so much rewriting?

Reply:
You can't copy it as is because `config.mk` has `a=b` while `xcp_inventory_config.ml` has `let a="b"`. I can simply undo it and print out directly the `xcp_inventory_config.ml` as it was originally, but the reason why I rewrote it was that I (a) wanted to avoid creating files in the source repo and (b) make it similar to other repos where we put configuration in a `config.mk` file.

Reply:
I would rather make the config thing write the correct content: `let a="b"`, and then echo it in the file or copy it as you prefer, but without all that manual string manipulation